theme/src/components/details.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import styled from 'styled-components'
+
+// The <details> element is not yet supported in Edge so we have to use a polyfill.
+// We have to check if window is defined before importing the polyfill
+// so the code doesn’t run while Gatsby is building.
+if (typeof window !== 'undefined') {
+  import('details-element-polyfill')
```

Comment:
Yes:
> import() calls use promises internally. If you use import() with older browsers, remember to shim Promise using a polyfill such as es6-promise or promise-polyfill.

https://webpack.js.org/guides/code-splitting/#dynamic-imports

Reply:
It's not ideal but we have to check that `window` is defined before importing because the polyfill assumes `window` is defined. If we import `details-element-polyfill` unconditionally it will break the Gatsby build because `window` is not defined at that time. 

I'm not too worried about this dynamic import because Edge is the only browser that needs this polyfill and I'm assuming that Edge users are a very small percentage of our traffic.